Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -23,6 +23,7 @@ ComponentInfo::ComponentInfo(
     boost::shared_ptr<const std::vector<size_t>> parentIndices,
     boost::shared_ptr<std::vector<Eigen::Vector3d>> positions,
     boost::shared_ptr<std::vector<Eigen::Quaterniond>> rotations,
+    boost::shared_ptr<std::vector<Eigen::Vector3d>> scaleFactors,
```

Comment:
@martyngigg basically this is the problem where `ParameterMaps` have been used to store generic information that is used by the Framework. We need to purge that information at the start from the `ParameterMap` so we have to assume it could exist.